build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -202,7 +202,7 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
                 List<string> cmakeArgList = new List<string>();
 
                 string cmakeBaseRid, visualStudio, archMacro, arch;
-                string ridMacro = $"-DCLI_CMAKE_RUNTIME_ID:STRING={rid}";
+                string ridMacro = $"-DCLI_CMAKE_RUNTIME_ID:STRING=win10-{platform.ToLower()}";
```

Comment:
The buildcontext variable I left out for the reasons mentioned in the code comment and because it would only ever be used in the corehost compilation and even then only for the Windows build. We've already got a few different RID variables in `CompileTargets.cs`; adding another to only be used in the win corehost build would only add more indirection. We've already got the other baseRIDs for Windows hardcoded based on the platform (cmakeBaseRid), so it's preferable having this calculation alongside the other ones.

I'm against renaming or moving any existing variables, at least as a part of this PR. I've been burned too many times by Core-Setup